mkdocs_simple_plugin/plugin.md

```diff
@@ -14,25 +14,114 @@ _Python 3.x, 3.5, 3.6, 3.7, 3.8, 3.9 supported._
 
 ```yaml
 site_name: your_site_name
 plugins:
   - simple:
-      # Optional setting to only include specific folders
+      # Only directories whose name (final path component) matches a "glob"
+      # pattern in this list will be searched for documentation:
       include_folders: ["*"]
-      # Optional setting to ignore specific folders
-      ignore_folders: [""]
-      # Optional setting to specify if hidden folders should be ignored
+      # In addition, directories whose name matches a glob pattern in this
+      # list will NOT be searched for documentation.
+      ignore_folders: ["site"]
+      # Hidden directories will not be searched if this is true:
       ignore_hidden: True
-      # Optional setting to specify other extensions besides md files to be copied
+      # If the following is true, the contents of the docs directory (if any)
+      # will be merged at the same level as all other documentation, i.e., the
+      # docs directory prefix of these files will be elided. Otherwise, the docs
+      # directory will be searched like all other directories, i.e., the docs
+      # directory prefix will be kept:
+      merge_docs_dir: True
+      # Any file in the searched directories whose name contains a string in
+      # this list will simply be copied to the generated documentation:
       include_extensions: [".tif", ".tiff", ".gif", ".jpeg", ".jpg", ".jif", ".jfif",
             ".jp2", ".jpx", ".j2k", ".j2c", ".fpx", ".pcd", ".png", ".pdf", "CNAME"]
-      # Optional setting to merge the docs directory with other documentation
-      merge_docs_dir: True
+      # The dicts in the following list specify which files will be scanned for
+      # embedded documentation and how. The first entry in the list gives all
+      # of the keys allowed in an entry with their meanings and default values.
+      # The remaining entries in the list actually constitute the default value
+      # of the "semiliterate" parameter:
+      semiliterate:
+          # Any file in the searched directories whose name contains this
+          # required regular expression parameter will be scanned:
+        - pattern: None
+          # However, if the file has already matched one of the
+          # "include_extensions" or an earlier semiliterate pattern, it will not
+          # be scanned again unless this parameter is true:
+          rescan: false
```

Comment:
Correct, that's why I'd like both the rescan and the to_extension parameters: so I can grab "user manual" and "api documentation" markdown from the same file, using two different extensions. Is it OK to leave in?

Reply:
I think it might be cleaner to specify a "folder" instead of an extension to put the files in per pattern then? I imagine having an "API documentation" folder in the output could be pretty useful.

Also, why not always rescan?

Reply:
I was just using a to_extension value of "_api.md", so that "module.py" could produce both "module.md" (from the "regular" scan) and a "module_api.md". That seemed to work pretty well.

Being able to turn off rescan is helpful when you have mixed types of source files, say Python, C, and Java. You want a `(.py)$` pattern to grab """-strings, and want the fallback /** .. **/ pattern for both the C and Java files, so it has a permissive pattern `(.[^.])*`. You probably don't want to rescan the Python files for /** .. **/ -- it won't ever be useful, and in the worst case a trigger might inadvertently match some string in your Python code --  so you set rescan false on the fallback pattern.

So here are your options as I see them:

(1) On to_extension, I could 
  (A) change the parameter name to "destination" and have it be a template to be expanded by the filename match, which if not specified simply replaces the entire match with ".md". This would allow what I am currently doing, plus allow your quite reasonable proposed behavior of putting api rescans into a separate folder:
```
- pattern: '^(.*)\.py$'
  destination: 'api/\1.md'
```

  (2) I could leave to_extension as is, a string to replace the last matching group of the pattern;

  (3) Or I could eliminate to_extension altogether, which would seem to preclude double-extraction from the same file. If you think double-extraction is worthwhile for `simple` to support but you don't like either (1) or (2), I'm open to any alternate suggestion for how to support it.

(B) On rescan, I could
  (1) Leave it as is,
  (2) Remove this parameter and behave as if always "true" -- the user will simply have to ensure that at most one of the extraction triggers occurs in any file that matches multiple patterns.
  (3) Remove this parameter and behave as if always "false" -- like (A3) this would seem to preclude double extraction but I am open to other suggestions of how to support it if you would like to.

Let me know what you prefer.

Reply:
(A)  
   (1) `pattern`/`destination` works for me.  `to_extension` makes me think you can export to`.txt` or `.pdf` or something, which it doesn't do.
(B) 
   (2) Pattern matches should occur on all files, all the time.  People should be prescriptive about what extensions they want to include for the pattern matching, and they could include the same type multiple times if they want to do some advanced configuration.